File: leetcode/solutions/0540.SingleElementInASortedArray/singleNonDuplicate.go
package solution

func singleNonDuplicate(nums []int) int {
	len := len(nums)

	for i := 0; i < len; i += 2 {
		if i == len-1 || nums[i] != nums[i+1] {
			return nums[i]
		}
	}

	return -1
}

File: leetcode/solutions/0540.SingleElementInASortedArray/singleNonDuplicate.ts
export default function singleNonDuplicate(nums: number[]): number {
  for (let i = 0; i < nums.length; i += 2) {
    if (i === nums.length - 1 || nums[i] !== nums[i + 1]) {
      return nums[i];
    }
  }

  return -1;
}
